### ISSUE TITLE
Drawers not locking properly/ghost items

### ISSUE THREAD

**Issue by Paril** [jaquadro/StorageDrawers#616]
So, this bug appears to be happening in Modern Skyblock 2. Not sure if another mod is causing it; I tried with both ExU Transfer Pipes and Thermal Ducts so far.

Here's some videos demonstrating the behavior:
https://streamable.com/tfkr3
https://streamable.com/7n52b

The drawer controller seems to be the catalyst here, but I don't understand why. The two drawer sets are not connected in any way other than transfer nodes. The second set will randomly get dirty gems "teleported" into them, but they're not real items and are just ghosts. If this happens, the slot becomes permanently unusable. I can't put stuff in them; on the client-side it appears as if I had, but even on the server-side I can't even pull it back out. It's still in my inventory. If I shift-rightclick or lock a drawer, it forcefully refreshes and shows the real contents... which sometimes still have ghosts in them, even if they're unlocked.

**Comment by jaquadro** [comment]
Could you send me your world?  It's a little hard to follow the videos alone, so it would be helpful if I could see this reproduce.

**Comment by Paril** [comment]
I'll send it over. You need the latest version of Modern Skyblock 2 to play with it; the drawers in question are downstairs, with a set of itemducts going from the crushers to the drawers, drawers to cleaners, cleaners to drawers.

http://paril.ca/2018-01-24-12-06-19.zip

The actual issue with the ghosts being there in the first place (not counting the issue where locking or shift-right-click will "reset" the ghosts temporarily, even if they're technically still there on server-side) is definitely caused by ItemDucts. I tried with Logistical Pipes from Mekanism and they worked fine. I put back ItemDucts, and within minutes the ghosts of the dirty ores were somehow transferring from the input ducts (the ones at the bottom-left; note they are not touching or interacting in any way with the outputs on the right) to the drawers they are nowhere near connected to. *Note that these dirty ores don't even actually exist in the drawers!* They are completely visual only, but still on both server & client-side.

![](https://i.imgur.com/BgFRa3Q.png)

If you shift-right-click or lock/unlock a drawer, the "ghosts" of the dirty ores will clear themselves. If you try to place items in these ghost spots, they will show as if they've gone in, but they'll still be in your inventory. The only way to fix them is to re-place the actual drawers and hope the ducts nearby don't suck a ghost into them.

Locking a drawer, in general, doesn't work properly either. If you lock a drawer, place an item in it, then take it back out, shift-rightclicking/locking will also act as if it was a ghost item and "delete" it visually, even though the locked item is still technically there on the server-side (pipes will pipe them into the locked slot no problem).

EDIT: Here's an even more odd thing:
![](https://i.imgur.com/Rf4yrHL.png)
I have no itemducts or anything transferring Stone around. The importers from RS are importing Stone into the Rock Crushers, but the ducts only suck out the output. Somehow the drawers got the ghost of Stone in them, despite Stone never moving in the pipe network. You can also see here how The One Probe shows that the drawer I'm looking at only has two items in it, but it has three visually; one of them is a ghost, and will disappear if I shift-rightclick.

**Comment by Paril** [comment]
Okay, I just realized that those ghost slots are actually server-side taken too. Even if I make them "disappear" by shift-rightclicking, the controller won't accept new items, so the slot is indeed filled despite not being able to pull anything out of that slot.

Also: https://streamable.com/riflw

**Comment by CJBok** [comment]
Today I discovered the same issue, I've created a pull request for the fix. #624

**Comment by jaquadro** [comment]
Can you test version 5.3.5?

**Comment by Routhinator** [comment]
I'm seeing this issue as well and we are on 5.3.6 - still happening. Only way to work around it is to break the drawer.
